### PR TITLE
[LLVM][Maintainers] Move code owner list to Maintainers.md

### PR DESCRIPTION
**Pull request by nikic**
Following https://discourse.llvm.org/t/rfc-proposing-changes-to-the-community-code-ownership-policy/80714 and https://github.com/llvm/llvm-project/pull/107384, this PR moves the llvm CODE_OWNERS.txt to the Maintainers.md file.

The general format of the Maintainers.md file follows the current CodeOwners.rst from clang, with some minor terminology changes to account for the code owners -> maintainers migration.

This change is intended to be a pure format change, without changing any actual code owners / maintainers. The list is still very badly outdated. The intent here is to get the format change out of the way, so we can get proper diffs when replacing no longer active maintainers. If we mix both together, it will be hard to follow what actually changed.

I've tried to categorize things as best as I could, and filled in GitHub handles for people where I knew them or could determine them with some confidence.

[**Rendered version of the file**](https://github.com/nikic/llvm-project/blob/llvm-maintainers/llvm/Maintainers.md)

### CODE REVIEW

**Comment by nikic** [comment]
Adding folks who reviewed https://github.com/llvm/llvm-project/pull/107384 to this one.

**Review by MaskRay — APPROVED** [review]

File: llvm/Maintainers.rst [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+~~~~~~~~~~~~~~~
+| Sundeep Kushwaha
+| sundeepk\@quicinc.com (email)
+
+Lanai backend
+~~~~~~~~~~~~~
+| Jacques Pienaar
+| jpienaar\@google.com (email), jpienaar
+
+Loongarch backend
```
Comment (topperc) [inline llvm/Maintainers.rst:209]: The directory name is spelled LoongArch

Comment (nikic) [reply llvm/Maintainers.rst]: Fixed!

**Review by rengolin — APPROVED** [review]

File: llvm/Maintainers.rst [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+
+
+Transforms and analyses
+-----------------------
+
+AliasAnalysis
+~~~~~~~~~~~~~~
+| Hal Finkel
+| hfinkel\@anl.gov (email), hfinkel (GitHub)
```
Comment (rnk) [inline llvm/Maintainers.rst:36]: Can you linkify the github names? I think you do this with a big regex, something like:
`s/, ([^ ]*) \(GitHub\)/, [\1](https://github.com/\1) (GitHub)/`
(untested)

**Review by rnk — APPROVED** [review]

File: llvm/Maintainers.rst (hunk 2 above)
Comment (nikic) [reply llvm/Maintainers.rst]: I've pushed the change for that, but I'm not sure I like the result. It does give us links in the rendered version, but it also makes the text file a good bit harder to read.

Comment (rnk) [reply llvm/Maintainers.rst]: For rst readability you could linkify the "GitHub" text. Also, sorry, as usual it's easy to get confused about when we're using rst or markdown. Since this is a new file, should we prefer Markdown, or is that premature?

Comment (nikic) [reply llvm/Maintainers.rst]: In https://llvm.org/docs/DeveloperPolicy.html#maintainers we mention Maintainers.rst, and this is what the clang and lldb CodeOwners.rst files currently use. Personally, I wouldn't mind switching this to markdown.

Comment (nikic) [reply llvm/Maintainers.rst]: I've pushed another commit converting the file to markdown, here's the result: https://github.com/nikic/llvm-project/blob/llvm-maintainers/llvm/Maintainers.md

Two things that were lost along the way is the table of contents, and the suppression of linkification for the emails (apparently this can't be cleanly disabled with markdown).

**Review by vitalybuka — APPROVED** [review]
